developer-docs/bolt_server.md

```diff
@@ -0,0 +1,147 @@
+# Bolt Server 
+
+## Overview
+Bolt server provides a service to run bolt tasks over SSH and WinRM. This service is exposed via an API as described in this document. 
+
+## Configuration
+Bolt server can be configured by defining content in HOCON format at one of the following expected configuration file path locations.
+
+**Global Config**: `/etc/puppetlabs/bolt-server/conf.d/bolt-server.conf`
+
+**Local Config**: `~/.puppetlabs/bolt-server.conf`
+
+**Options**
+- `port`: Integer, *optional* - The port the bolt server will run on (default 8144)
+- `ssl-cert`: String, *required* - Path to the cert file.
+- `ssl-key`: String, *required* - Path to the key file.
+- `ssl-ca-cert`: String, *required* - Path to the ca cert file.
+
+**Example**
+```
+bolt-server: {
+    port: 8144
+    ssl-cert: /etc/puppetlabs/bolt-server/ssl/cert.pem
+    ssl-key: /etc/puppetlabs/bolt-server/ssl/private_key.pem
+    ssl-ca-cert: /etc/puppetlabs/bolt-server/ssl/ca.pem
+} 
+```
+
+## API Endpoints
+Each API endpoint accepts a request as described below. The request body must be a JSON object.
+
+## POST /ssh/run_task
+- `target`: [SSH Target Object](#ssh-target-object), *required* - Target information to run task on.
+- `task`: [Task Object](#task-object), *required* - Task to run on target.
+- `parameters`: Object, *optional* - JSON formatted parameters to be provided to task.
+
+### SSH Target Object
+The Target be a JSON object. The following keys are available:
+- `hostname`: String, *required* - Target identifier.
+- `password`: String, *optional* - Password for SSH transport authentication.
+- `port`: Integer, *optional* - Connection port (Default: `22`).
+- `user`: String, *optional* - Login user (Default: `root`).
+- `connect-timeout`: Integer, *optional* - How long Bolt should wait when establishing connections.
+- `run-as-command`: Array, *optional* - Command elevate permissions. Bolt appends the user and command strings to the configured `run-as` command before running it on the target. This command must not require an interactive password prompt, and the `sudo-password` option is ignored when `run-as-command` is specified.
+- `run-as`: String, *optional* - A different user to run commands as after login.
+- `tmpdir`: String, *optional* - The directory to upload and execute temporary files on the target.
+- `host-key-check`: Bool, *optional* - Whether to perform host key validation when connecting over SSH (Default: `true`).
+- `sudo-password`: String, *optional* - Password to use when changing users via `run-as`
+
+### Task Object
+The Task be a JSON object. The following keys are available:
+- `name`: String, *required* - The name of the task.
+- `metadata`: String, *optional* - The contents of the task's metadata.json file. 
+- `file`: String, *required* - Task's base64 encoded file content.
+
+## Response 
+If the task runs the response will have status 200.
+The response will be a standard bolt Result JSON object.
+
+## Example
```

Comment:
This should be something like "Development" or "Testing bolt changes" - the service is available in PE now, and all this setup is only really worth it if you want to use a pe-bolt-server that isn't in PE builds.